ably/rest_client.go

```diff
@@ -353,17 +359,11 @@ func (c *RestClient) doWithHandle(r *Request, handle func(*http.Response, interf
 	if err != nil {
 		log.Error("RestClient: error handling response: ", err)
 		if e, ok := err.(*Error); ok {
-			if canFallBack(e.StatusCode) &&
-				(c.opts.FallbackHostsUseDefault ||
-					strings.HasPrefix(req.URL.Host, defaultOptions.RestHost) ||
-					c.opts.FallbackHosts != nil) {
-				fallback := defaultOptions.FallbackHosts
-				if c.opts.FallbackHosts != nil {
-					fallback = c.opts.FallbackHosts
-				}
-				log.Info("RestClient: trying to fallback with hosts=%v", fallback)
-				if len(fallback) > 0 {
-					left := fallback
+			if canFallBack(e.StatusCode) {
+				fallbacks, _ := c.opts.getFallbackHosts()
+				log.Info("RestClient: trying to fallback with hosts=%v", fallbacks)
+				if fallbacks != nil && len(fallbacks) > 0 {
```

Comment:
This `nil` check on `fallbacks` is redundant, since the length of a nil slice is zero so is covered by `len(fallbacks) > 0`

Reply:
Okay, the whitespace issue is resolved by formatter.  I didn't intend to introduce it.

Reply:
Fixed the issue